sdk/go/telemetry/span.go

```diff
@@ -43,3 +43,12 @@ func End(span trace.Span, fn func() error) {
 	}
 	span.End()
 }
+
+// EndNow is equivalent to End, but doesn't take a function.
+func EndNow(span trace.Span, err error) {
```

Comment:
I do like this though. Also thought about changing `End` to take a `*error` as its second arg, instead of a whole closure, which should be enough for the same basic use case (capturing `rerr`). But both are a little crufty.

Reply:
Well I'll keep this bit :)